### PR TITLE
Rewrite Project

### PR DESCRIPTION
**Pull request by alexmercerind**
## TL;DR

I want to feel satisfied about the project & the work I've done.

<hr>

I want architecture & implementation to be in a good shape. This calls for a major rewrite (and/or) refactor. The project has been in development for a good amount of time & over the time many compromises have been made:

1. Many libraries & tooling was nonexistent previously. This lead to workarounds & hacks at multiple places e.g. caching media library into JSON. The media library has now been rewritten e.g. to be multi-threaded & utilize SQLite i.e.  [drift](https://github.com/simolus3/drift). This has simplified the business logic at multiple places. Similarly, migrating to newer packages is one task e.g. [go_router](https://github.com/flutter/packages/blob/main/packages/go_router).
2. A lot of implementation has been open-sourced as independent projects e.g. most notably [media_kit](https://github.com/media-kit/media-kit) (really proud of my work in it, both architecture & implementation). It is important for the project to follow the same level of quality as the work previously derived from it.
3. UI code is duplicated at multiple places e.g. different widgets for all screens e.g. AlbumScreen, ArtistScreen, PlaylistScreen etc. even though they are relatively the same. This was previously done without much of a second thought, however it is indeed increasing the amount of time required to make any changes e.g. 4.
4. [Material Design 3](https://m3.material.io/) has been released recently. It will be a good decision to refactor project so that both Material Design 3 & Material Design 2 are supported and any future theme changes can be made extensively.
5. macOS support.
6. Unit-tests. It is important to ensure the correct behavior of the core modules such as media_library & tag_reader. Now a sufficient amount of unit-tests are in place to solidify the stability of the project.

There are many other reasons as well. During this period, I have also evolved as an engineer & have learnt a lot more about software development in general (now that I'm about to graduate). It's time to put those ideas into practice here as well.

### CODE REVIEW

**Comment by ryzendew** [comment]
can you add builds to actions please i would like to test and give feed back

**Comment by alexmercerind** [comment]
Thanks a lot for your interest @ryzendew. You may join our Discord to provide feedback, where I share updates etc.
